--- dnspython.orig/dns/dnssecalgs/dsa.py
+++ dnspython/dns/dnssecalgs/dsa.py
@@ -29,6 +29,8 @@ class PublicDSA(CryptographyPublicKey):
         dsa_t = (self.key.key_size // 8 - 64) // 8
         if dsa_t > 8:
             raise ValueError("unsupported DSA key size")
+        if pn.parameter_numbers.q > 2**160:
+            raise ValueError("unsupported qbit size")
         octets = 64 + dsa_t * 8
         res = struct.pack("!B", dsa_t)
         res += pn.parameter_numbers.q.to_bytes(20, "big")
