--- a/dns/message.py
+++ b/dns/message.py
@@ -852,7 +852,7 @@
         self.one_rr_per_rrset = \
             self.message._get_one_rr_per_rrset(self.one_rr_per_rrset)
         self._get_question(MessageSection.QUESTION, qcount)
-        if self.question_only:
+        if self.question_only or self.message.flags & dns.flags.TC:
             return
         self._get_section(MessageSection.ANSWER, ancount)
         self._get_section(MessageSection.AUTHORITY, aucount)
