--- a/dns/message.py
+++ b/dns/message.py
@@ -1179,7 +1179,7 @@
         )
         try:
             self._get_question(MessageSection.QUESTION, qcount)
-            if self.question_only:
+            if self.question_only or self.message.flags & dns.flags.TC:
                 return self.message
             self._get_section(MessageSection.ANSWER, ancount)
             self._get_section(MessageSection.AUTHORITY, aucount)
